### PR TITLE
thread,misc: add support for affinity

### PR DESCRIPTION
**Pull request by qdaoming-intel**
Back ported thread affinity feature and related dependency commits
from master. It will add support for those APIs: uv_cpumask_size, 
uv_thread_setaffinity,uv_thread_getaffinity,uv_thread_detach.
The supported platforms are Linux, FreeBSD, and Windows.
Empty implementations (returning -ENOTSUP) on non-supported platforms
(such as OS X and AIX).

### CODE REVIEW

**Comment by qdaoming-intel** [comment]
@vtjnash Please review here for back porting thread affinity feature related commits from master.

File: docs/src/threading.rst [hunk 1]
```diff
@@ -88,6 +88,45 @@ Threads
...
+    the bytes in cpumask to a bitmask and uses SetThreadAffinityMask() which
+    returns the previous affinity setting.
+
+    The mask_size specifies the number of entries (bytes) in cpumask / oldmask,
+    and must be greater-than-or-equal-to :c:func:`uv_cpumask_size`.
+
+    .. note::
+        Thread affinity setting is not atomic on Windows. Unsupported on macOS.
+
+    .. versionadded:: 2.0.0
```
Comment (vtjnash) [inline docs/src/threading.rst:107]: Change these all to 1.45.0

File: include/uv.h [hunk 2]
```diff
@@ -1782,6 +1783,14 @@ UV_EXTERN int uv_thread_create_ex(uv_thread_t* tid,
...
                                   void* arg);
+UV_EXTERN int uv_thread_setaffinity(uv_thread_t* tid,
+                                    char* cpumask,
+                                    char* oldmask,
+                                    size_t mask_size);
+UV_EXTERN int uv_thread_getaffinity(uv_thread_t* tid,
+                                    char* cpumask,
+                                    size_t mask_size);
+UV_EXTERN int uv_thread_detach(uv_thread_t* tid);
 UV_EXTERN uv_thread_t uv_thread_self(void);
```
[the remark this answers is not in the document]
Comment (qdaoming-intel) [reply include/uv.h:1794]: As the order is consistent with the latest status in master, I will not accept the suggested change.

**Comment by qdaoming-intel** [comment]
@vtjnash Are the 2 commits resolve your comments? If yes, could you please merge this PR?
Then I will create another PR for new features necessary for node.js affinity optimization.

**Review by vtjnash — APPROVED** [review]

**Comment by qdaoming-intel** [comment]
@vtjnash  As this PR is approved 2 days ago, could you please merge it?

**Comment by qdaoming-intel** [comment]
@vtjnash @bnoordhuis 
My next PR will depend on this PR's merging. Do you know when this PR will be merged?

**Pull request merged**